src/components/Dropdown.tsx

```diff
@@ -0,0 +1,103 @@
+            }
+        };
+
+        document.addEventListener("mousedown", handleClickOutside);
+
+        return () => {
+            document.removeEventListener("mousedown", handleClickOutside);
+        };
+    }, [dropdownRef]);
+
+    return (
+        // Dropdown component container
+        <div id={id + "-component"} className="relative p-5 max-w-sm mx-auto">
+            <h3 className="p-4 text-lg font-semibold">{title}</h3>
+            {/* The dropdown element */}
+            <div id={id} ref={dropdownRef} className="relative inline-block w-64">
+                <button
+                    aria-label="Toggle dropdown"
+                    aria-haspopup="true"
+                    aria-expanded={isOpen}
+                    type="button"
+                    onClick={() => setIsOpen(!isOpen)}
+                    className="w-full flex justify-between items-center px-4 py-2 text-gray-900 dark:text-white bg-white dark:bg-gray-800 border border-gray-300 dark:border-gray-600 rounded-md shadow-sm focus:outline-none focus:ring-2 focus:ring-blue-500"
+                >
+                    {options[selectedIndex]}
+                    <span className="ml-2">▼</span>
+                </button>
+                {/* The dropdown menu */}
+                {isOpen && (
+                    <div aria-label="Dropdown menu">
+                        <ul role="menu" className="absolute left-0 mt-1 w-full bg-white dark:bg-gray-800 border border-gray-300 dark:border-gray-600 rounded-md shadow-lg">
+                            {options.map((option, index) => (
+                                <li
+                                    key={option}
+                                    className="px-4 py-2 hover:bg-gray-100 dark:hover:bg-gray-700 cursor-pointer"
+                                    onClick={() => { handleSelection(index) }}
+                                    >
+                                    {option}
+                                </li>
+                            ))}
+                        </ul>
+                    </div>
+                )}
+            </div>
+        </div>
+    )
+}
+
+/**
+ * Represents the objects that the Dropdown component accepts as props.
+ * @type {Object} DropdownProps
+ * @property {string} id - The id of the dropdown, used to differentiate between different dropdowns.
+ * @property {string} title - The header of the dropdown component, typically the instructions.
+ * @property {string[]} options - The array of options to be displayed in the dropdown menu.
+ * @property {number} selectedIndex - The starting element index to display in the dropdown menu.
+ * @property {Dispatch<SetStateAction<number>>} setSelectedIndex - The parent component's "set state"
+ *          method for handling option selection.
+ */
+type DropdownProps = {
+    id: string;
+    title: string;
+    options: string[];
+    selectedIndex: number;
+    setSelectedIndex: Dispatch<SetStateAction<number>>;
+}
```

Comment:
Since this type is only for the `Dropdown` component, I didn't think it was worth adding to the type.ts file. Happy to change if there are differing opinions!